src/core/execute.c

```diff
@@ -361,9 +360,13 @@ static int connect_logger_as(
                 !!context->syslog_level_prefix,
                 is_syslog_output(output),
                 is_kmsg_output(output),
-                is_terminal_output(output));
+                is_terminal_output(output)) < 0)
+                return -errno;
 
-        return move_fd(fd, nfd, false);
+        if (dprintf(fd, " ") < 0)
+                return -errno;
+
+        return move_fd(TAKE_FD(fd), nfd, false);
 }
 static int open_terminal_as(const char *path, int flags, int nfd) {
```

Comment:
somehow the empty line above this one went missing, might make sense to add it back in with this commit, even if it's slightly unrelated.